main.js

```diff
@@ -1,48 +1,75 @@
 import * as THREE from 'three';
+import { OrbitControls } from 'three/addons/controls/OrbitControls.js';
 const loader = new THREE.TextureLoader();
 
 const scene = new THREE.Scene();
-const camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 0.1, 1000 );
-
 const renderer = new THREE.WebGLRenderer();
+
 renderer.setSize( window.innerWidth, window.innerHeight );
 document.body.appendChild( renderer.domElement );
+const boardSize = 14; // Including the edges
+
+const grid = new THREE.GridHelper(boardSize, boardSize);
+grid.position.set(0, 0, 0);
+grid.rotation.x = Math.PI / 2; // rotate 90 degrees
+scene.add(grid);
+
+// Add ambient light
+const ambientLight = new THREE.AmbientLight(0xffffff, 0.5);
+scene.add(ambientLight);
+
+// Add directional light (like sunlight)
+const directLight = new THREE.DirectionalLight(0xffffff, 0.5);
+directLight.position.set(0, 10, 10);
+scene.add(directLight);
+
+const camera = new THREE.PerspectiveCamera( 75, window.innerWidth / window.innerHeight, 0.1, 1000 );
+camera.position.set(0, 0, boardSize);
+camera.lookAt(0, 0, 0);
 
-// Loading and applying the texture
-loader.load('public/pandav-tank-88-KbSZbuEc-unsplash.jpg', function ( texture ) {
-    const material = new THREE.MeshBasicMaterial( { map: texture, side: THREE.DoubleSide } );
+const raycaster = new THREE.Raycaster();
+const mouse = new THREE.Vector2();
 
-    const boardSize = 10;
-    let halfSize = boardSize / 2;
-    let boardPlaneGeometry = new THREE.PlaneGeometry(boardSize, boardSize);
+window.addEventListener('click', onClick, false);
 
-    // Add the game board as a simple plane first.
-    let boardPlane = new THREE.Mesh(boardPlaneGeometry, material);
-    boardPlane.rotation.x = Math.PI / 2;
-    boardPlane.position.set(halfSize, -0.01, halfSize);
+function onClick(event) {
```

Comment:
Typically these functions are called "handlers", i.e. `clickHandler`. Normally, we try to reserve `onClick` for the event name.